builder/swig/go.bzl

```diff
@@ -0,0 +1,174 @@
+        headers = depset(swig_headers, transitive = [lib_compilation_context.headers]),
+        defines = lib_compilation_context.defines,
+        framework_includes = lib_compilation_context.framework_includes,
+        includes = depset(
+            [h.dirname for h in ctx.attr.lib[CcInfo].compilation_context.headers.to_list()],
+            transitive = [lib_compilation_context.includes],
+        ),
+        local_defines = lib_compilation_context.local_defines,
+        quote_includes = lib_compilation_context.quote_includes,
+        system_includes = depset(
+            [],
+            transitive = [lib_compilation_context.system_includes],
+        )
+    )
+
+    return [
+        DefaultInfo(files = depset([wrap_src])),
+        OutputGroupInfo(
+            go_source = depset([wrap_go_dir]),
+            cxx_source = depset([wrap_src]),
+        ),
+        CcInfo(
+            compilation_context = compilation_context,
+            linking_context = ctx.attr.lib[CcInfo].linking_context,
+        ),
+    ]
+
+_swig_go_wrapper = rule(
+    implementation = _swig_go_wrapper_impl,
+    attrs = {
+        "lib": attr.label(
+            doc = "The C library for which to generate the wrapper",
+            providers = [CcInfo],
+            mandatory = True,
+        ),
+        "interface": attr.label(
+            doc = "Optional SWIG interface (.i) file",
+            allow_single_file = True,
+        ),
+        "includes": attr.label_list(
+            doc = "Additional SWIG files required for wrapper generation",
+            allow_files = True,
+        ),
+        "package_name": attr.string(
+            doc = "Package name override for generated Go package",
+        ),
+        "enable_cxx": attr.bool(
+            doc = "Enable SWIG C++ processing (default: False)",
+            default = False,
+        ),
+        "extra_args": attr.string_list(
+            doc = "Extra arguments to be passed to SWIG",
+        ),
+        "_swig": attr.label(
+            default = Label("@swig//:swig"),
+            allow_single_file = True,
+            executable = True,
+            cfg = "host",
+        ),
+    },
+)
+
+def swig_go(name, lib, package_name, shared_lib_name=None, tags=[], **kwargs):
+    swig_wrapper_name = name + "_swig"
+
+    _swig_go_wrapper(
+        name = swig_wrapper_name,
+        lib = lib,
+        package_name=package_name,
+        **kwargs,
+    )
+
+    if not shared_lib_name:
+        shared_lib_name = name
+
+    native.filegroup(
+        name = "source_file",
+        srcs = [swig_wrapper_name],
+        output_group = "go_source",
+    )
+
+    # TODO windows would use copy, maybe use an if statement or select to check
+    cmd = "cp $(SRCS)/{pkg_name}.go  $@".format(pkg_name=package_name)
+
+    # Copy the generated go file into an output file with the same name.
+    native.genrule(
+        name="go_wrapper_copier",
+        outs= [package_name + ".go"],
```

Comment:
I saw your reply on my oudated comment, but have we tested that `go_library` doesn't work with directories? Do we `need` this copier?

Reply:
I tried but struggled with getting bazel to find the file, by using `native.glob(["go/*.go"])` and similar but it finds an empty file and is undefined. Might be worth looking into but currently the genrule is working